overlord/snapstate/export_test.go

```diff
@@ -269,8 +269,12 @@ var (
 	MissingDisabledServices = missingDisabledServices
 )
 
-func (m *SnapManager) MaybeUndoRemodelBootChanges(t *state.Task) error {
-	return m.maybeUndoRemodelBootChanges(t)
+func (m *SnapManager) MaybeUndoRemodelBootChanges(t *state.Task) (restartRequested, rebootRequired bool, err error) {
+	restartPoss, err := m.maybeUndoRemodelBootChanges(t)
```

Comment:
(nitpick) I wonder if we should just be closer to the non-exported code here. It would feel more "naturally" to me. Something like the diff below maybe?
```go
diff --git a/overlord/snapstate/export_test.go b/overlord/snapstate/export_test.go
index 2fcbfc88a7..4265490350 100644
--- a/overlord/snapstate/export_test.go
+++ b/overlord/snapstate/export_test.go
@@ -269,12 +269,10 @@ var (
        MissingDisabledServices = missingDisabledServices
 )
 
-func (m *SnapManager) MaybeUndoRemodelBootChanges(t *state.Task) (restartRequested, rebootRequired bool, err error) {
-       restartPoss, err := m.maybeUndoRemodelBootChanges(t)
-       if restartPoss != nil {
-               return true, restartPoss.RebootRequired, nil
-       }
-       return false, false, err
+type RestartPossibility = restartPossibility
+
+func (m *SnapManager) MaybeUndoRemodelBootChanges(t *state.Task) (*RestartPossibility, error) {
+       return m.maybeUndoRemodelBootChanges(t)
 }
 
 func MockPidsOfSnap(f func(instanceName string) (map[string][]int, error)) func() {
diff --git a/overlord/snapstate/handlers_link_test.go b/overlord/snapstate/handlers_link_test.go
index a28aad2528..d13e1a74cb 100644
--- a/overlord/snapstate/handlers_link_test.go
+++ b/overlord/snapstate/handlers_link_test.go
@@ -1944,9 +1944,9 @@ func (s *linkSnapSuite) TestMaybeUndoRemodelBootChangesUnrelatedAppDoesNothing(c
                },
        })
 
-       restartRequested, _, err := s.snapmgr.MaybeUndoRemodelBootChanges(t)
+       restartPos, err := s.snapmgr.MaybeUndoRemodelBootChanges(t)
        c.Assert(err, IsNil)
-       c.Check(restartRequested, Equals, false)
+       c.Check(restartPos, IsNil)
 }
 
 func (s *linkSnapSuite) TestMaybeUndoRemodelBootChangesSameKernel(c *C) {
@@ -1963,9 +1963,9 @@ func (s *linkSnapSuite) TestMaybeUndoRemodelBootChangesSameKernel(c *C) {
                Type: "kernel",
        })
 
-       restartRequested, _, err := s.snapmgr.MaybeUndoRemodelBootChanges(t)
+       restartPos, err := s.snapmgr.MaybeUndoRemodelBootChanges(t)
        c.Assert(err, IsNil)
-       c.Check(restartRequested, Equals, false)
+       c.Check(restartPos, IsNil)
 }
 
 func (s *linkSnapSuite) TestMaybeUndoRemodelBootChangesNeedsUndo(c *C) {
@@ -2017,7 +2017,7 @@ func (s *linkSnapSuite) TestMaybeUndoRemodelBootChangesNeedsUndo(c *C) {
        })
 
        // now we simulate that the new kernel is getting undone
-       restartRequested, rebootRequired, err := s.snapmgr.MaybeUndoRemodelBootChanges(t)
+       restartPos, err := s.snapmgr.MaybeUndoRemodelBootChanges(t)
        c.Assert(err, IsNil)
 
        // that will schedule a boot into the previous kernel
@@ -2026,8 +2026,8 @@ func (s *linkSnapSuite) TestMaybeUndoRemodelBootChangesNeedsUndo(c *C) {
                "snap_kernel":     "kernel_1.snap",
                "snap_try_kernel": "",
        })
-       c.Check(restartRequested, Equals, true)
-       c.Check(rebootRequired, Equals, true)
+       c.Assert(restartPos, NotNil)
+       c.Check(restartPos.RebootRequired, Equals, true)
 }
 
 func (s *linkSnapSuite) testDoLinkSnapWithToolingDependency(c *C, classicOrBase string) {
```